edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -141,11 +142,47 @@ public async Task SendEventAsync(Message message)
             }
         }
 
-        internal static void ValidateSchemaVersion(string schemaVersion)
+        internal static void ValidateSchemaVersion(DeploymentConfig config)
         {
-            if (ExpectedSchemaVersion.CompareMajorVersion(schemaVersion, "desired properties schema") != 0)
+            string schemaVersion = config.SchemaVersion;
+
+            if (string.IsNullOrWhiteSpace(schemaVersion) || !Version.TryParse(schemaVersion, out Version actualSchemaVersion))
+            {
+                throw new InvalidSchemaVersionException($"Invalid desired properties schema version {schemaVersion}");
+            }
+
+            // Check major version and upper bound
+            if (actualSchemaVersion.Major != ExpectedSchemaVersion.Major ||
+                actualSchemaVersion.Major > ExpectedSchemaVersion.Major)
             {
-                Events.MismatchedMinorVersions(schemaVersion, ExpectedSchemaVersion);
+                throw new InvalidSchemaVersionException($"The desired properties schema version {schemaVersion} is not compatible with the expected version {ExpectedSchemaVersion}");
+            }
+
+            // Validate minor versions
+            if (actualSchemaVersion.Minor == 0)
+            {
+                // 1.0
+                //
+                // Module startup order is not supported
+                foreach (KeyValuePair<string, IModule> kvp in config.Modules)
+                {
+                    IModule moduleConfig = kvp.Value;
+
+                    if (moduleConfig.StartupOrder != Constants.DefaultPriority)
+                    {
+                        throw new InvalidSchemaVersionException($"Module startup order is not supported in schema {actualSchemaVersion}.");
```

Comment:
Why do we need to throw here? Maybe log and ignore? Also, this doesn't seem to be a sustainable way to handle adding new properties to the schema. I feel like we need to come up with a generic way to handle this.